plugins/modem-manager/fu-mm-device.c

```diff
@@ -1494,6 +1500,31 @@ fu_mm_device_set_progress(FuDevice *self, FuProgress *progress)
 	fu_progress_add_step(progress, FWUPD_STATUS_DEVICE_BUSY, 2);	/* reload */
 }
 
+static gboolean
+fu_mm_device_set_quirk_kv(FuDevice *device,
+			  const gchar *key,
+			  const gchar *value,
+			  GError **error)
+{
+	FuMmDevice *self = FU_MM_DEVICE(device);
+
+	/* load from quirks */
+	if (g_strcmp0(key, "ModemManagerDetachAtFastbootHasResponse") == 0) {
+		if (g_strcmp0(value, "true") == 0)
+			self->detach_at_fastboot_has_response = TRUE;
```

Comment:
This should be a flag, e.g. `fu_device_register_private_flag()` and then `Flags = detach-at-fastboot-has-response`

Reply:
I tried that (updated the commits), but for some reason these flags are also not processed...
I checked other plugins and I don't see what I'm missing to be honest.

Any advice?

Reply:
It should be coming from `fu_device_set_quirk_kv`->`fu_device_set_custom_flags`->`fu_device_set_custom_flag`->`fu_device_private_flag_item_find_by_str`

Two ways to debug it:
1. Share verbose fwupdtool output, we can see if it's listed there.
2. Set a breakpoint in `fu_device_set_custom_flags` and run fwupdtool get-devices `--plugins=fastboot --plugins-modem-manager` (hopefully I got that right) and walk through it.

Reply:
Ah this shows up: `18:28:34:0551 FuDevice             no registered custom flag detach-at-fastboot-has-response on FuUsbDevice`

**Full log**

```
pine64-pinephone:/home/dylan# fwupdtool --verbose get-devices
18:28:33:0886 FuDebug              Verbose debugging enabled (on console 1)
18:28:33:0888 FuMain               locked /run/lock/fwupdtool
18:28:33:0889 FuConfig             loading config values from /etc/fwupd/daemon.conf
18:28:33:0889 FuConfig             using autodetected max archive size 18.4 EB
18:28:33:0891 GLib-GIO             _g_io_module_get_default: Found default implementation local (GLocalVfs) for ‘gio-vfs’
18:28:33:0893 FuRemoteList         loading /usr/share/fwupd/metainfo
18:28:33:0894 XbSilo               attempting to load /var/cache/fwupd/metainfo.xmlb
18:28:33:0894 XbSilo               file: 8a8e3c95-b816-9f6c-1ff6-543d49ad5e0a, current:8a8e3c95-b816-9f6c-1ff6-543d49ad5e0a, cached: (null)
18:28:33:0894 XbSilo               loading silo with file contents
18:28:33:0895 FuRemoteList         loading remote from /etc/fwupd/remotes.d/vendor.conf
18:28:33:0896 FuRemoteList         loading remote from /etc/fwupd/remotes.d/lvfs.conf
18:28:33:0897 FuRemoteList         loading remote from /etc/fwupd/remotes.d/lvfs-testing.conf
18:28:33:0899 FuRemoteList         loading remote from /etc/fwupd/remotes.d/vendor-directory.conf
18:28:33:0899 FuRemoteList         loading remote from /etc/fwupd/remotes.d/fwupd-tests.conf
18:28:33:0901 FuRemoteList         ignoring unfound remote fwupd
18:28:33:0901 FuRemoteList         ordering lvfs-testing=lvfs+1
18:28:33:0901 Jcat                 ignoring GPG-KEY-Linux-Foundation-Firmware as not PKCS-7 certificate
18:28:33:0901 Jcat                 ignoring GPG-KEY-Linux-Vendor-Firmware-Service as not PKCS-7 certificate
18:28:33:0901 Jcat                 trying to load certificate from /etc/pki/fwupd/LVFS-CA.pem
18:28:33:0901 Jcat                 reading /etc/pki/fwupd/LVFS-CA.pem with 1679 bytes
18:28:33:0903 Jcat                 loaded 1 certificates
18:28:33:0905 Jcat                 ignoring GPG-KEY-Linux-Foundation-Metadata as not PKCS-7 certificate
18:28:33:0906 Jcat                 ignoring GPG-KEY-Linux-Vendor-Firmware-Service as not PKCS-7 certificate
18:28:33:0906 Jcat                 trying to load certificate from /etc/pki/fwupd-metadata/LVFS-CA.pem
18:28:33:0907 Jcat                 reading /etc/pki/fwupd-metadata/LVFS-CA.pem with 1679 bytes
18:28:33:0908 Jcat                 loaded 1 certificates
18:28:33:0908 Jcat                 reading /var/lib/fwupd/pki/secret.key with 2459 bytes
18:28:33:0908 Jcat                 reading /var/lib/fwupd/pki/client.pem with 1383 bytes
18:28:33:0908 FuEngine             client certificate now exists: nothing to do
18:28:33:0908 FuHistory            trying to open database '/var/lib/fwupd/pending.db'
18:28:33:0911 FuHistory            got schema version of 7
18:28:33:0912 FuHwids              loading HwId overrides from /etc/fwupd/daemon.conf
18:28:33:0912 FuHwids              not loading HwId overrides from /var/lib/fwupd/daemon.conf
18:28:33:0913 FuHwids              ignoring Manufacturer: no data available
18:28:33:0916 FuHwids              smbios property EnclosureKind=8
18:28:33:0918 FuHwids              ignoring Family: offset bigger than size 6
18:28:33:0918 FuHwids              smbios property ProductName=Pine64 PinePhone (1.2)
18:28:33:0918 FuHwids              ignoring ProductSku: offset bigger than size 6
18:28:33:0918 FuHwids              ignoring BiosVendor: offset bigger than size 0
18:28:33:0918 FuHwids              ignoring BiosVersion: offset bigger than size 0
18:28:33:0919 FuHwids              ignoring BiosMajorRelease: offset bigger than size 0
18:28:33:0919 FuHwids              ignoring BiosMinorRelease: offset bigger than size 0
18:28:33:0919 FuHwids              ignoring FirmwareMajorRelease: offset bigger than size 0
18:28:33:0919 FuHwids              ignoring FirmwareMinorRelease: offset bigger than size 0
18:28:33:0919 FuHwids              ignoring BaseboardManufacturer: offset bigger than size 0
18:28:33:0919 FuHwids              ignoring BaseboardProduct: offset bigger than size 0
18:28:33:0919 FuHwids              HardwareID-0 is not available, not available as 'Manufacturer' unknown
18:28:33:0919 FuHwids              HardwareID-1 is not available, not available as 'Manufacturer' unknown
18:28:33:0919 FuHwids              HardwareID-2 is not available, not available as 'Manufacturer' unknown
18:28:33:0919 FuHwids              HardwareID-3 is not available, not available as 'Manufacturer' unknown
18:28:33:0919 FuHwids              HardwareID-4 is not available, not available as 'Manufacturer' unknown
18:28:33:0919 FuHwids              HardwareID-5 is not available, not available as 'Manufacturer' unknown
18:28:33:0920 FuHwids              HardwareID-6 is not available, not available as 'Manufacturer' unknown
18:28:33:0920 FuHwids              HardwareID-7 is not available, not available as 'Manufacturer' unknown
18:28:33:0920 FuHwids              HardwareID-8 is not available, not available as 'Manufacturer' unknown
18:28:33:0920 FuHwids              HardwareID-9 is not available, not available as 'Manufacturer' unknown
18:28:33:0920 FuHwids              HardwareID-10 is not available, not available as 'Manufacturer' unknown
18:28:33:0920 FuHwids              HardwareID-11 is not available, not available as 'Manufacturer' unknown
18:28:33:0920 FuHwids              HardwareID-12 is not available, not available as 'Manufacturer' unknown
18:28:33:0920 FuHwids              HardwareID-13 is not available, not available as 'Manufacturer' unknown
18:28:33:0920 FuHwids              HardwareID-14 is not available, not available as 'Manufacturer' unknown
18:28:33:0921 FuEngine             building metadata for remote 'vendor-directory'
18:28:33:0922 FuEngine             building metadata for /usr/share/fwupd/remotes.d/vendor/firmware/EG25GGBR07A08M2G_01.002.01.002-recovery.cab
18:28:33:0961 FuCommon             reading /usr/share/fwupd/remotes.d/vendor/firmware/EG25GGBR07A08M2G_01.002.01.002-recovery.cab with 11644303 bytes
18:28:33:0961 FuEngine             Emitting PropertyChanged('Status'='decompressing')
Decompressing…           [-                                      ]18:28:34:0131 FuCabinet            failed to verify EG25GGBR07A08M2G_01.002.01.002.metainfo.xml: no JcatItem
18:28:34:0131 FuCabinet            processing file: EG25GGBR07A08M2G_01.002.01.002.metainfo.xml
18:28:34:0131 XbSilo               compiling 6040e4d37707646f5576d1e9c4be58c2400ebe39:filename=EG25GGBR07A08M2G_01.002.01.002-recovery\EG25GGBR07A08M2G_01.002.01.002.metainfo.xml:prefix=components…
18:28:34:0133 FuCabinet            processing release: EG25GGBR07A08M2G_01.002.01.002
18:28:34:0187 FuEngine             Emitting PropertyChanged('Status'='idle')
Decompressing…           [***************************************]
18:28:34:0196 FuEngine             ignoring: /usr/share/fwupd/remotes.d/vendor/firmware/README.md
18:28:34:0196 XbSilo               attempting to load /var/cache/fwupd/metadata.xmlb
18:28:34:0196 XbSilo               file: e311220e-9159-3ce8-2a44-4ee8cb9cee68, current:e311220e-9159-3ce8-2a44-4ee8cb9cee68, cached: (null)
18:28:34:0197 XbSilo               loading silo with file contents
18:28:34:0203 FuEngine             1204 components now in silo
18:28:34:0246 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_modem_manager.so)
18:28:34:0248 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_parade_lspcon.so)
18:28:34:0248 FuContext            added udev subsystem watch of i2c
18:28:34:0249 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_nvme.so)
18:28:34:0249 FuContext            added udev subsystem watch of nvme
18:28:34:0250 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_ebitdo.so)
18:28:34:0252 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_uefi_capsule.so)
18:28:34:0253 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_synaptics_cxaudio.so)
18:28:34:0254 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_fresco_pd.so)
18:28:34:0255 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_uefi_pk.so)
18:28:34:0256 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_colorhug.so)
18:28:34:0257 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_ata.so)
18:28:34:0257 FuContext            added udev subsystem watch of block
18:28:34:0258 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_elanfp.so)
18:28:34:0260 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_logitech_hidpp.so)
18:28:34:0260 FuContext            added udev subsystem watch of hidraw
18:28:34:0261 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_analogix.so)
18:28:34:0262 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_jabra.so)
18:28:34:0264 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_dfu.so)
18:28:34:0265 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_ccgx.so)
18:28:34:0266 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_linux_lockdown.so)
18:28:34:0266 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_optionrom.so)
18:28:34:0267 FuContext            added udev subsystem watch of pci
18:28:34:0269 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_logitech_bulkcontroller.so)
18:28:34:0270 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_nordic_hid.so)
18:28:34:0271 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_elantp.so)
18:28:34:0271 FuContext            added udev subsystem watch of i2c-dev
18:28:34:0272 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_rts54hid.so)
18:28:34:0273 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_powerd.so)
18:28:34:0274 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_rts54hub.so)
18:28:34:0275 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_iommu.so)
18:28:34:0275 FuContext            added udev subsystem watch of iommu
18:28:34:0276 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_dfu_csr.so)
18:28:34:0277 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_emmc.so)
18:28:34:0278 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_realtek_mst.so)
18:28:34:0279 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_cpu.so)
18:28:34:0280 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_hailuck.so)
18:28:34:0281 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_linux_tainted.so)
18:28:34:0289 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_tpm.so)
18:28:34:0289 FuContext            added udev subsystem watch of tpm
18:28:34:0291 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_bcm57xx.so)
18:28:34:0292 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_fastboot.so)
18:28:34:0293 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_dell_dock.so)
18:28:34:0295 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_ep963x.so)
18:28:34:0296 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_steelseries.so)
18:28:34:0297 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_synaptics_cape.so)
18:28:34:0298 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_acpi_phat.so)
18:28:34:0299 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_synaptics_mst.so)
18:28:34:0299 FuContext            added udev subsystem watch of drm
18:28:34:0300 FuContext            added udev subsystem watch of drm_dp_aux_dev
18:28:34:0301 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_uefi_recovery.so)
18:28:34:0302 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_nitrokey.so)
18:28:34:0303 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_wacom_raw.so)
18:28:34:0304 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_vli.so)
18:28:34:0305 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_pci_bcr.so)
18:28:34:0306 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_goodixmoc.so)
18:28:34:0308 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_superio.so)
18:28:34:0309 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_thelio_io.so)
18:28:34:0309 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_system76_launch.so)
18:28:34:0310 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_pci_mei.so)
18:28:34:0312 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_synaptics_prometheus.so)
18:28:34:0313 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_thunderbolt.so)
18:28:34:0313 FuContext            added udev subsystem watch of thunderbolt
18:28:34:0314 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_pixart_rf.so)
18:28:34:0316 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_upower.so)
18:28:34:0316 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_linux_swap.so)
18:28:34:0319 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_synaptics_rmi.so)
18:28:34:0319 FuContext            added udev subsystem watch of serio
18:28:34:0320 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_logind.so)
18:28:34:0321 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_wacom_usb.so)
18:28:34:0322 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_uefi_dbx.so)
18:28:34:0324 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_cros_ec.so)
18:28:34:0325 FuPlugin             init(/usr/lib/fwupd-plugins-5/libfu_plugin_mtd.so)
18:28:34:0325 FuContext            added udev subsystem watch of mtd
18:28:34:0325 FuPluginList         uefi_capsule [0] to be ordered after upower [0] so promoting to [1]
18:28:34:0325 FuPluginList         uefi_recovery [0] to be ordered after uefi_capsule [1] so promoting to [2]
18:28:34:0325 FuPluginList         cannot find plugin 'msr' requested by 'cpu'
18:28:34:0325 FuPluginList         bcm57xx [0] better than optionrom [0] so bumping to [1]
18:28:34:0325 FuPluginList         cannot find plugin 'msr' requested by 'cpu'
18:28:34:0325 FuPluginList         dell_dock [0] better than synaptics_mst [0] so bumping to [1]
18:28:34:0325 FuPluginList         cannot find plugin 'msr' requested by 'cpu'
18:28:34:0361 XbSilo               attempting to load /var/cache/fwupd/quirks.xmlb
18:28:34:0362 XbSilo               file: a16ffa3d-ab7f-6bfa-0dbb-e84ce0d1b6f0, current:a16ffa3d-ab7f-6bfa-0dbb-e84ce0d1b6f0, cached: (null)
18:28:34:0362 XbSilo               loading silo with file contents
18:28:34:0368 FuContext            battery threshold now 10
18:28:34:0368 FuEngine             Emitting PropertyChanged('Status'='loading')
Loading…                 [ -                                     ]18:28:34:0369 FuPlugin             startup(bios)
18:28:34:0370 FuPlugin             startup(lenovo_thinklmi)
18:28:34:0370 FuEngine             disabling plugin because: failed to startup using lenovo_thinklmi: thinklmi not available
18:28:34:0370 FuPlugin             startup(linux_lockdown)
18:28:34:0371 FuPlugin             startup(linux_swap)
18:28:34:0371 FuPlugin             startup(linux_tainted)
18:28:34:0372 FuPlugin             startup(logind)
18:28:34:0382 FuPlugin             startup(logitech_hidpp)
18:28:34:0382 FuPlugin             startup(modem_manager)
18:28:34:0399 FuPlugin             startup(mtd)
18:28:34:0399 FuPlugin             startup(powerd)
18:28:34:0405 FuEngine             disabling plugin because: failed to startup using powerd: no service that owns the name for org.chromium.PowerManager
18:28:34:0405 FuPlugin             startup(thunderbolt)
18:28:34:0405 FuPlugin             startup(tpm)
18:28:34:0406 FuPlugin             startup(upower)
18:28:34:0414 FuContext            battery state now charging
18:28:34:0414 FuContext            battery level now 99
18:28:34:0415 FuPlugin             startup(uefi_capsule)
18:28:34:0415 FuEngine             disabling plugin because: failed to startup using uefi_capsule: offset bigger than size 0
18:28:34:0415 FuPlugin             startup(uefi_recovery)
18:28:34:0415 FuEngine             disabling plugin because: failed to startup using uefi_recovery: kernel efivars support missing: /sys/firmware/efi/efivars
18:28:34:0415 FuPlugin             coldplug(acpi_phat)
18:28:34:0416 FuEngine             disabling plugin because: failed to coldplug using acpi_phat: Failed to open file “/sys/firmware/acpi/tables/PHAT”: No such file or directory
18:28:34:0416 FuPlugin             coldplug(amt)
18:28:34:0416 FuEngine             disabling plugin because: failed to coldplug using amt: Unable to find a ME interface
18:28:34:0416 FuPlugin             coldplug(bios)
18:28:34:0416 FuPlugin             coldplug(cpu)
18:28:34:0417 FuEngine             disabling plugin because: failed to coldplug using cpu: no <cpuid.h> support
18:28:34:0417 FuPlugin             coldplug(modem_manager)
18:28:34:0423 FuPluginMm           ModemManager 1.18.2 is available
18:28:34:0424 FuPluginMm           valid firmware update combination: fastboot
18:28:34:0426 FuDevice             no FuContext assigned for FuMmDevice:
  Name:                 QUECTEL Mobile Broadband Module
  Summary:              Mobile broadband device
  Protocol:             com.google.fastboot
  Flags:                updatable|use-runtime-version
  Vendor:               QUALCOMM INCORPORATED
  Version:              EG25GGBR07A08M2G - 05010820
  VersionFormat:        plain
  Icon:                 network-modem
  PhysicalId:           /sys/devices/platform/soc/1c1b000.usb/usb2/2-1
  InternalFlags:        replug-match-guid
  AtPort:               /dev/ttyUSB2

18:28:34:0426 FuDevice             no FuContext assigned for FuMmDevice:
  Name:                 QUECTEL Mobile Broadband Module
  Guid:                 c33a4560-8681-55b6-bbb6-85258f2de149 ← USB\VID_2C7C&PID_0125&REV_0318&CARRIER_ROW_GENERIC_3GPP ⚠
  Summary:              Mobile broadband device
  Protocol:             com.google.fastboot
  Flags:                updatable|use-runtime-version
  Vendor:               QUALCOMM INCORPORATED
  Version:              EG25GGBR07A08M2G - 05010820
  VersionFormat:        plain
  Icon:                 network-modem
  PhysicalId:           /sys/devices/platform/soc/1c1b000.usb/usb2/2-1
  InternalFlags:        replug-match-guid
  AtPort:               /dev/ttyUSB2

18:28:34:0426 FuDevice             no FuContext assigned for FuMmDevice:
  Name:                 QUECTEL Mobile Broadband Module
  Guid:                 c33a4560-8681-55b6-bbb6-85258f2de149 ← USB\VID_2C7C&PID_0125&REV_0318&CARRIER_ROW_GENERIC_3GPP ⚠
  Guid:                 1a2996cb-f86e-5583-a464-e1b96e1c6ae9 ← USB\VID_2C7C&PID_0125&REV_0318 ⚠
  Summary:              Mobile broadband device
  Protocol:             com.google.fastboot
  Flags:                updatable|use-runtime-version
  Vendor:               QUALCOMM INCORPORATED
  Version:              EG25GGBR07A08M2G - 05010820
  VersionFormat:        plain
  Icon:                 network-modem
  PhysicalId:           /sys/devices/platform/soc/1c1b000.usb/usb2/2-1
  InternalFlags:        replug-match-guid
  AtPort:               /dev/ttyUSB2

18:28:34:0427 FuDevice             no FuContext assigned for FuMmDevice:
  Name:                 QUECTEL Mobile Broadband Module
  Guid:                 c33a4560-8681-55b6-bbb6-85258f2de149 ← USB\VID_2C7C&PID_0125&REV_0318&CARRIER_ROW_GENERIC_3GPP ⚠
  Guid:                 1a2996cb-f86e-5583-a464-e1b96e1c6ae9 ← USB\VID_2C7C&PID_0125&REV_0318 ⚠
  Guid:                 587bf468-6859-5522-93a7-6cce552a0aa3 ← USB\VID_2C7C&PID_0125 ⚠
  Summary:              Mobile broadband device
  Protocol:             com.google.fastboot
  Flags:                updatable|use-runtime-version
  Vendor:               QUALCOMM INCORPORATED
  Version:              EG25GGBR07A08M2G - 05010820
  VersionFormat:        plain
  Icon:                 network-modem
  PhysicalId:           /sys/devices/platform/soc/1c1b000.usb/usb2/2-1
  InternalFlags:        replug-match-guid
  AtPort:               /dev/ttyUSB2

18:28:34:0427 FuDevice             using 976c4a39e87f61e6940ea6a8d39c583cfa99615f for /sys/devices/platform/soc/1c1b000.usb/usb2/2-1
18:28:34:0427 FuPlugin             emit added from modem_manager: 976c4a39e87f61e6940ea6a8d39c583cfa99615f
18:28:34:0479 XbSilo               ignoring for OR statement: element name suggests is unknown in silo
18:28:34:0482 FuPlugin             fu_plugin_device_registered(dell_dock)
18:28:34:0483 FuPlugin             fu_plugin_device_registered(pci_bcr)
18:28:34:0483 FuPlugin             fu_plugin_device_registered(thunderbolt)
18:28:34:0483 FuPlugin             fu_plugin_device_registered(tpm)
18:28:34:0483 FuPlugin             fu_plugin_device_registered(uefi_pk)
18:28:34:0483 FuDeviceList         ::added 976c4a39e87f61e6940ea6a8d39c583cfa99615f
18:28:34:0483 FuMain               ADDED:
FuMmDevice:
  DeviceId:             976c4a39e87f61e6940ea6a8d39c583cfa99615f
  Name:                 QUECTEL Mobile Broadband Module
  Guid:                 c33a4560-8681-55b6-bbb6-85258f2de149 ← USB\VID_2C7C&PID_0125&REV_0318&CARRIER_ROW_GENERIC_3GPP
  Guid:                 1a2996cb-f86e-5583-a464-e1b96e1c6ae9 ← USB\VID_2C7C&PID_0125&REV_0318
  Guid:                 587bf468-6859-5522-93a7-6cce552a0aa3 ← USB\VID_2C7C&PID_0125
  Guid:                 22ae45db-f68e-5c55-9c02-4557dca238ec ← USB\VID_2C7C
  Summary:              Mobile broadband device
  Plugin:               modem_manager
  Protocol:             com.google.fastboot
  Flags:                updatable|supported|registered|use-runtime-version
  Vendor:               QUALCOMM INCORPORATED
  VendorId:             USB:0x2C7C
  Version:              EG25GGBR07A08M2G - 05010820
  VersionFormat:        plain
  Icon:                 network-modem
  Created:              2021-12-13
  PhysicalId:           /sys/devices/platform/soc/1c1b000.usb/usb2/2-1
  InternalFlags:        replug-match-guid
  AtPort:               /dev/ttyUSB2

18:28:34:0525 XbSilo               ignoring for OR statement: element name suggests is unknown in silo
18:28:34:0527 FuPlugin             coldplug(superio)
18:28:34:0528 FuPlugin             coldplug(tpm)
18:28:34:0528 FuPluginTpm          failed to load eventlog: Failed to open file “/sys/kernel/security/tpm0/binary_bios_measurements”: No such file or directory
18:28:34:0528 FuPlugin             coldplug(uefi_dbx)
18:28:34:0528 FuDevice             using 230c8b18-8d9b-53ec-838b-6cfc0383493a for main-system-firmware
18:28:34:0529 FuEngine             disabling plugin because: failed to coldplug using uefi_dbx: Error opening file /sys/firmware/efi/efivars/KEK-8be4df61-93ca-11d2-aa0d-00e098032b8c: No such file or directory
18:28:34:0529 FuPlugin             coldplug(uefi_pk)
18:28:34:0529 FuEngine             disabling plugin because: failed to coldplug using uefi_pk: Error opening file /sys/firmware/efi/efivars/PK-8be4df61-93ca-11d2-aa0d-00e098032b8c: No such file or directory
18:28:34:0529 FuEngine             using plugins: acpi_dmar, acpi_facp, analogix, ata, bcm57xx, bios, ccgx, colorhug, cros_ec, dell_dock, dfu, dfu_csr, ebitdo, elanfp, elantp, emmc, ep963x, fastboot, fresco_pd, goodixmoc, hailuck, iommu, jabra, linux_lockdown, linux_sleep, linux_swap, linux_tainted, logind, logitech_bulkcontroller, logitech_hidpp, modem_manager, mtd, nitrokey, nordic_hid, nvme, optionrom, parade_lspcon, pci_bcr, pci_mei, pixart_rf, realtek_mst, rts54hid, rts54hub, steelseries, superio, synaptics_cape, synaptics_cxaudio, synaptics_mst, synaptics_prometheus, synaptics_rmi, system76_launch, thelio_io, thunderbolt, tpm, upower, vli, wacom_raw, wacom_usb
18:28:34:0551 FuDevice             no registered custom flag detach-at-fastboot-has-response on FuUsbDevice
18:28:34:0748 FuEngine             failed to add device /sys/devices/platform/soc/1c0f000.mmc/mmc_host/mmc0/mmc0:1234/block/mmcblk0: failed get ffu_capable for (null)
18:28:34:0772 FuDevice             using 0e0c93c8b4bb222157feedbde8f863e23bd1a8f7 for DEVPATH=/devices/platform/soc/1c11000.mmc/mmc_host/mmc2/mmc2:0001
18:28:34:0777 FuPlugin             emit added from emmc: 0e0c93c8b4bb222157feedbde8f863e23bd1a8f7
18:28:34:0809 XbSilo               ignoring for OR statement: indexed string '8c6f96d4-f406-53fa-84ca-a367cd368dfc' was unfound
18:28:34:0810 XbSilo               ignoring for OR statement: indexed string 'add792f2-7053-5638-80e7-20987fd16769' was unfound
18:28:34:0811 FuPlugin             fu_plugin_device_registered(dell_dock)
18:28:34:0811 FuPlugin             fu_plugin_device_registered(pci_bcr)
18:28:34:0811 FuPlugin             fu_plugin_device_registered(thunderbolt)
18:28:34:0811 FuPlugin             fu_plugin_device_registered(tpm)
18:28:34:0811 FuDeviceList         ::added 0e0c93c8b4bb222157feedbde8f863e23bd1a8f7
18:28:34:0812 FuMain               ADDED:
FuEmmcDevice:
  DeviceId:             0e0c93c8b4bb222157feedbde8f863e23bd1a8f7
  Name:                 DA4032
  Guid:                 8c6f96d4-f406-53fa-84ca-a367cd368dfc ← EMMC\DA4032
  Guid:                 add792f2-7053-5638-80e7-20987fd16769 ← EMMC\0069&0256
  Guid:                 5500994c-1b73-52b4-acf4-bb574b6f3029 ← EMMC\0069&0256&DA4032
  Serial:               0xb2161c15
  Plugin:               emmc
  Protocol:             org.jedec.mmc
  Flags:                internal|updatable|registered
  VendorId:             EMMC:0x000045
  Version:              0x3034313430363139
  VersionFormat:        number
  Icon:                 media-memory
  Created:              2021-12-13
  PhysicalId:           DEVPATH=/devices/platform/soc/1c11000.mmc/mmc_host/mmc2/mmc2:0001
  BackendId:            /sys/devices/platform/soc/1c11000.mmc/mmc_host/mmc2/mmc2:0001/block/mmcblk2
  PossiblePlugin:       ata
  PossiblePlugin:       emmc
  InternalFlags:        is-open
  SysfsPath:            /sys/devices/platform/soc/1c11000.mmc/mmc_host/mmc2/mmc2:0001/block/mmcblk2
  Subsystem:            block
  DeviceFile:           /dev/mmcblk2
  SectorSize:           512

18:28:34:0813 XbSilo               ignoring for OR statement: indexed string '8c6f96d4-f406-53fa-84ca-a367cd368dfc' was unfound
18:28:34:0813 XbSilo               ignoring for OR statement: indexed string 'add792f2-7053-5638-80e7-20987fd16769' was unfound
18:28:34:0923 FuEngine             Emitting PropertyChanged('Status'='idle')
Loading…                 [***************************************]
18:28:34:0925 XbSilo               ignoring for OR statement: indexed string '8c6f96d4-f406-53fa-84ca-a367cd368dfc' was unfound
18:28:34:0925 XbSilo               ignoring for OR statement: indexed string 'add792f2-7053-5638-80e7-20987fd16769' was unfound
18:28:34:0967 XbSilo               ignoring for OR statement: element name suggests is unknown in silo
18:28:34:0970 XbSilo               ignoring for OR statement: indexed string '8c6f96d4-f406-53fa-84ca-a367cd368dfc' was unfound
18:28:34:0970 XbSilo               ignoring for OR statement: indexed string 'add792f2-7053-5638-80e7-20987fd16769' was unfound
18:28:35:0012 XbSilo               ignoring for OR statement: element name suggests is unknown in silo
18:28:35:0014 FuEngine             writing motd target /var/cache/fwupd/motd.d/85-fwupd
WARNING: UEFI capsule updates not available or enabled in firmware setup
  See https://github.com/fwupd/fwupd/wiki/PluginFlag:capsules-unsupported for more information.
WARNING: This package has not been validated, it may not work properly.
18:28:35:0016 FuMain               FuEmmcDevice:
  DeviceId:             0e0c93c8b4bb222157feedbde8f863e23bd1a8f7
  Name:                 DA4032
  Guid:                 8c6f96d4-f406-53fa-84ca-a367cd368dfc ← EMMC\DA4032
  Guid:                 add792f2-7053-5638-80e7-20987fd16769 ← EMMC\0069&0256
  Guid:                 5500994c-1b73-52b4-acf4-bb574b6f3029 ← EMMC\0069&0256&DA4032
  Serial:               0xb2161c15
  Plugin:               emmc
  Protocol:             org.jedec.mmc
  Flags:                internal|updatable|registered
  VendorId:             EMMC:0x000045
  Version:              0x3034313430363139
  VersionFormat:        number
  Icon:                 media-memory
  Created:              2021-12-13

18:28:35:0016 FuMain               FuMmDevice:
  DeviceId:             976c4a39e87f61e6940ea6a8d39c583cfa99615f
  Name:                 QUECTEL Mobile Broadband Module
  Guid:                 c33a4560-8681-55b6-bbb6-85258f2de149 ← USB\VID_2C7C&PID_0125&REV_0318&CARRIER_ROW_GENERIC_3GPP
  Guid:                 1a2996cb-f86e-5583-a464-e1b96e1c6ae9 ← USB\VID_2C7C&PID_0125&REV_0318
  Guid:                 587bf468-6859-5522-93a7-6cce552a0aa3 ← USB\VID_2C7C&PID_0125
  Guid:                 22ae45db-f68e-5c55-9c02-4557dca238ec ← USB\VID_2C7C
  Summary:              Mobile broadband device
  Plugin:               modem_manager
  Protocol:             com.google.fastboot
  Flags:                updatable|supported|registered|use-runtime-version
  Vendor:               QUALCOMM INCORPORATED
  VendorId:             USB:0x2C7C
  Version:              EG25GGBR07A08M2G - 05010820
  VersionFormat:        plain
  Icon:                 network-modem
  Created:              2021-12-13

18:28:35:0023 FuPlugin             destroy(synaptics_mst)
18:28:35:0023 FuPlugin             destroy(logind)
18:28:35:0024 FuPlugin             destroy(pci_mei)
18:28:35:0024 FuPlugin             destroy(upower)
18:28:35:0025 FuPlugin             destroy(modem_manager)
18:28:35:0028 FuPlugin             destroy(linux_lockdown)
18:28:35:0028 FuPlugin             destroy(linux_swap)
18:28:35:0029 FuPlugin             destroy(powerd)
18:28:35:0030 FuPlugin             destroy(tpm)
18:28:35:0031 FuPlugin             destroy(linux_tainted)
18:28:35:0032 FuPlugin             destroy(uefi_capsule)
```